sdk/iOS/src/MSURLBuilder.m

```diff
@@ -253,11 +253,12 @@ +(NSString *) queryStringFromQuery:(MSQuery *)query
 
 
 #pragma mark * Private Methods
-
-
 // This is for 'strict' URL encoding that will encode even reserved URL
 // characters.  It should be used only on URL pieces, not full URLs.
 NSString* encodeToPercentEscapeString(NSString *string) {
+    //Solves deprecation issue, but fails the tests. Must study it.
+//    return [string stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet characterSetWithCharactersInString:@"!*;:@&=+/?%#[]"]];
+
```

Comment:
The meaning has flipped, so those are not encoded, you want something closer to: [string stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLQueryAllowedCharacterSet]];